File: 368-minABSDiffOfTwoNumbers.py

import math


def minABSDiffOfTwoNumbers(nums):
    nums.sort()
    minDiff=math.inf
    for i in range(1,len(nums)):
        minDiff=min(nums[i]-nums[i-1],minDiff)
    ans=[]
    for i in range(1,len(nums)):
        curDiff=nums[i]-nums[i-1]
        if curDiff==minDiff:
            ans+=[[nums[i-1],nums[i]]]
    return ans

def minAbsDiffOfTwoNumbers(nums):
    sz=len(nums)
    ans=[]
    minAbs=math.inf
    for i in range(sz):
        for j in range(sz):
            if i==j:
                continue
            if abs(nums[j]-nums[i])<minAbs:
                minAbs=abs(nums[j]-nums[i])
                
    for i in range(sz-1):
        for j in range(i,sz):
            if i==j:
                continue
            curDiff=abs(nums[i]-nums[j])
            if curDiff==minAbs:
                ans+=[[nums[i],nums[j]]]
                
    return ans



nums=[3,4,100,30,20,19,21]
print(minABSDiffOfTwoNumbers(nums))

File: 368-minABSDiffOfTwoNumbers.cpp
#include<vector>
#include<iostream>
#include<algorithm>

using VEC=std::vector<int>;


std::vector<VEC> minABSDiffOfTwoNumbers(VEC& nums)
{
    std::vector<VEC> ans{};
    std::sort(nums.begin(),nums.end());

    int minDiff=INT_MAX;

    for(int i=1;i<nums.size();++i)
    {
        minDiff=std::min(nums[i]-nums[i-1],minDiff);
    }

    for(int i=1;i<nums.size();++i)
    {
        int curMin=nums[i]-nums[i-1];
        if(curMin==minDiff)
        {
            ans.push_back(VEC{nums[i-1],nums[i]});
        }
    }
    return ans;
}

int main()
{
    VEC nums{3,4,100,30,20,19,21};
    std::vector<VEC> res=minABSDiffOfTwoNumbers(nums);
    for(auto _v:res)
    {
        std::cout<<'\n';
        for(auto v:_v)
        {
            std::cout<<v<<',';
        }
    }
    return 0;
}